### DISCUSSION TITLE
Building OT_COMISSIONER gives error

### DISCUSSION THREAD

**Discussion by guillermovidalsule** [openthread/openthread#9697] [Q&A]
I'm trying to build and configure the commissioner following this guide: https://openthread.io/guides/commissioner/build.md

However, when I execute the command `ninja -j1` I get the following error:

Notes
- The error might occur multiple times in different steps, it always looks quite similar.
- The execution does not stop because of this error as it proceeds to the next step, but at step `[21/253] Building CXX object src/app/cli/CMakeFiles/interpreter-test.dir/interpreter_test.cpp.o` it stays there for a very long time (>30min) and then it says:
`c++: fatal error: Killed signal terminated program cc1plus  
compilation terminated.  
ninja: build stopped: subcommand failed.
`
- When I stop the execution I have A LOT of these very similar errors + warnings.
- I'm using a BBB, being the image installed the one on the BBB guide provided by openthreads website.

![Screenshot from 2023-12-08 16-14-28](https://github.com/openthread/openthread/assets/133428133/a8dd2980-1736-48ff-842b-885f125831e3)

`[2/256] Building CXX object src/app/CMakeFiles/commissioner-app.dir/commissioner_app.cpp.o
../src/app/commissioner_app.cpp: In member function ‘virtual ot::commissioner::Error ot::commissioner::CommissionerApp::SetPanId(ot::commissioner::PanId, ot::commissioner::CommissionerApp::MilliSeconds)’:
../src/app/commissioner_app.cpp:690:7: note: parameter passing for argument of type ‘ot::commissioner::PanId’ changed in GCC 7.1
 Error CommissionerApp::SetPanId(PanId aPanId, MilliSeconds aDelay)
       ^~~~~~~~~~~~~~~
In file included from /usr/include/c++/8/map:60,
                 from ../src/app/commissioner_app.hpp:44,
                 from ../src/app/commissioner_app.cpp:34:
/usr/include/c++/8/bits/stl_tree.h: In member function ‘std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::size_type std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::erase(const _Key&) [with _Key = ot::commissioner::CommissionerApp::JoinerKey; _Val = std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>; _KeyOfValue = std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >; _Compare = std::less<ot::commissioner::CommissionerApp::JoinerKey>; _Alloc = std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >]’:
/usr/include/c++/8/bits/stl_tree.h:2518:7: note: parameter passing for argument of type ‘std::_Rb_tree<ot::commissioner::CommissionerApp::JoinerKey, std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>, std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >, std::less<ot::commissioner::CommissionerApp::JoinerKey>, std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> > >::const_iterator’ {aka ‘std::_Rb_tree_const_iterator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >’} changed in GCC 7.1
       _M_erase_aux(__p.first, __p.second);
       ^~~~~~~~~~~~
/usr/include/c++/8/bits/stl_tree.h: In member function ‘std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::iterator std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::erase(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::iterator) [with _Key = ot::commissioner::CommissionerApp::JoinerKey; _Val = std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>; _KeyOfValue = std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >; _Compare = std::less<ot::commissioner::CommissionerApp::JoinerKey>; _Alloc = std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >]’:
/usr/include/c++/8/bits/stl_tree.h:1141:2: note: parameter passing for argument of type ‘std::_Rb_tree<ot::commissioner::CommissionerApp::JoinerKey, std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>, std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >, std::less<ot::commissioner::CommissionerApp::JoinerKey>, std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> > >::const_iterator’ {aka ‘std::_Rb_tree_const_iterator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >’} changed in GCC 7.1
  _M_erase_aux(__position);
  ^~~~~~~~~~~~
/usr/include/c++/8/bits/stl_tree.h: In member function ‘void std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_erase_aux(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator, std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator) [with _Key = ot::commissioner::CommissionerApp::JoinerKey; _Val = std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>; _KeyOfValue = std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >; _Compare = std::less<ot::commissioner::CommissionerApp::JoinerKey>; _Alloc = std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >]’:
/usr/include/c++/8/bits/stl_tree.h:2500:5: note: parameter passing for argument of type ‘std::_Rb_tree<ot::commissioner::CommissionerApp::JoinerKey, std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>, std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >, std::less<ot::commissioner::CommissionerApp::JoinerKey>, std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> > >::const_iterator’ {aka ‘std::_Rb_tree_const_iterator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >’} changed in GCC 7.1
     _Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/8/bits/stl_tree.h:2500:5: note: parameter passing for argument of type ‘std::_Rb_tree<ot::commissioner::CommissionerApp::JoinerKey, std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>, std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >, std::less<ot::commissioner::CommissionerApp::JoinerKey>, std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> > >::const_iterator’ {aka ‘std::_Rb_tree_const_iterator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >’} changed in GCC 7.1
/usr/include/c++/8/bits/stl_tree.h:2507:4: note: parameter passing for argument of type ‘std::_Rb_tree_const_iterator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >’ changed in GCC 7.1
    _M_erase_aux(__first++);
    ^~~~~~~~~~~~
/usr/include/c++/8/bits/stl_tree.h: In member function ‘void std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_erase_aux(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator) [with _Key = ot::commissioner::CommissionerApp::JoinerKey; _Val = std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>; _KeyOfValue = std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >; _Compare = std::less<ot::commissioner::CommissionerApp::JoinerKey>; _Alloc = std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >]’:
/usr/include/c++/8/bits/stl_tree.h:2486:5: note: parameter passing for argument of type ‘std::_Rb_tree<ot::commissioner::CommissionerApp::JoinerKey, std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo>, std::_Select1st<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >, std::less<ot::commissioner::CommissionerApp::JoinerKey>, std::allocator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> > >::const_iterator’ {aka ‘std::_Rb_tree_const_iterator<std::pair<const ot::commissioner::CommissionerApp::JoinerKey, ot::commissioner::JoinerInfo> >’} changed in GCC 7.1
     _Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
`

I don't even know where to start to fix this error, please help me.

Thank you!

**Comment by wgtdkp** [comment]
> I'm using a BBB, being the image installed the one on the BBB guide provided by openthreads website.

Hi, what's "BBB"?

Could you provide the complete output of the build command in an attached txt file?

**Comment by guillermovidalsule** [comment]
Hello!
- BBB means BeagleBone Black
- The image I used is `bone-debian-10.3-iot-armhf-2020-04-06-4gb.img.xz` which is the one mentioned in the BBB guide
- The board is connected to a 5V power supply and via ethernet to the router
- The RCP is an nRF52840 dongle

According to https://stackoverflow.com/questions/48149323/what-does-the-gcc-warning-project-parameter-passing-for-x-changed-in-gcc-7-1-m most of the output are warnings and can be ignored. But there are some errors and it ends up getting killed, usually at interpreter / interpreter_test if I'm not wrong.

I'm attaching the error message in a txt file:
[error.txt](https://github.com/openthread/openthread/files/13627878/error.txt)
